web-services/query/src/main/java/datawave/webservice/query/configuration/QueryData.java

```diff
@@ -2,43 +2,106 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
 
 import org.apache.accumulo.core.client.IteratorSetting;
 import org.apache.accumulo.core.data.Range;
-
-import com.google.common.collect.Lists;
-import com.google.common.collect.Sets;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 
 /**
  * Class to encapsulate all required information to run a query.
- *
  */
 public class QueryData {
-    List<IteratorSetting> settings = Lists.newArrayList();
-    String query;
-    Collection<Range> ranges = Sets.newHashSet();
-    Collection<String> columnFamilies = Sets.newHashSet();
+    private String query;
+    private Collection<Range> ranges = new HashSet<>();
+    private Collection<String> columnFamilies = new HashSet<>();
+    private List<IteratorSetting> settings = new ArrayList<>();
+    private int hashCode = -1;
```

Comment:
`-1` is technically a valid hashCode. Not the end of the world but if something legitimately hashed to -1, then we would keep recomputing. Typically you'd use `-1` when a negative number is not feasible - like for a count. In this case, the behavior works b/c we will just recompute it again but a more robust check would be to have a boolean for the need to recompute the hash.

Reply:
+1